### PR TITLE
Add wifi_connected primitive

### PR DESCRIPTION
**Pull request by beardhatcode**
I wrote this without compiling or testing, this must be done before merging

#### TODO

- [x] Test this on a real device

### CODE REVIEW

File: examples/assemblyscript/main/smartlamp.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export function main(): void {
     // Connect to Wi-Fi
     until_connected(
         () => { wd.wifi_connect(SSID, PASSWORD); },
-        () => { return wd.wifi_status() == wd.WL_CONNECTED; });
```
Comment (beardhatcode) [inline examples/assemblyscript/main/smartlamp.ts:43]: @tolauwae , will this work?

File: examples/assemblyscript/main/smartlamp.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export function main(): void {
...
+        wd.wifi_connected);
     let message = "Connected to wifi network with ip: ";
```
Comment (beardhatcode) [inline examples/assemblyscript/main/smartlamp.ts:44]: @tolauwae , will this work?

Comment (tolauwae) [reply examples/assemblyscript/main/smartlamp.ts:44]: It should. I will test on an ESP tomorrow.

**Review by tolauwae — APPROVED** [review]
Works!

**Pull request merged**